be/src/util/block_compression.cpp

```diff
@@ -669,7 +669,7 @@ class SnappySlicesSource : public snappy::Source {
     // REQUIRES: Available() >= n
     void Skip(size_t n) override {
         _available -= n;
```

Comment:
1. this function has requirement for the caller:  `// REQUIRES: Available() >= n`, if the caller can not make sure this, maybe the caller have some bug;
2. After the change in this pr, `_available` may be corrupted;
3. If we really want to make `Skip` safe, I think only add a check at the begin of the function is better:
 ```if (unlikely(_available < n)) { return; }```

Reply:
when available = 0 and n = 0, the requirement is met.